### ISSUE TITLE
Overriding json tag from embedded struct not working 

### ISSUE THREAD

**Issue by JonasHiltl** [danielgtaylor/huma#299]
I have this struct which is used as a body

```go
type BodyA struct {
	A string `json:"a" doc:"share this"`
	B string `json:"b" doc:"specific"`
}
```
In a different endpoint I want to reuse the struct but hide a specific field, so that I can for example share the `doc` tag of the other fields, like this.
```go
type BodyB struct {
	BodyA
	B string `json:"-"`
}
```
But the generated openapi spec still shows the `b` field of `BodyB`, even though it should be hidden.

**Comment by victoraugustolls** [comment]
@JonasHiltl I opened PR #304 as a fix for this issue!

**Comment by danielgtaylor** [comment]
Fixed in #304, and will be released within the next few days. Thanks!
